tensorboard/data/server/logdir.rs

```diff
@@ -169,13 +97,11 @@ impl<'a> LogdirLoader<'a> {
     ///
     /// Panics if the `commit.runs` lock is poisoned.
     fn synchronize_runs(&mut self, discoveries: &Discoveries) {
-        let discoveries = &discoveries.0;
-
         // Remove runs with no event files. (This could be cleaner and more efficient with
         // `HashMap::drain_filter`, but that's not yet stabilized.)
         let mut removed: Vec<Run> = Vec::new();
         self.runs.retain(|run, _| {
-            if discoveries.contains_key(run) {
+            if discoveries.get(run).map_or(false, |fs| !fs.is_empty()) {
```

Comment:
Should empty `Vec` be an allowed state?  We might as well handle the fallback in any case, but maybe worth clarifying the contract above.

Reply:
Good eye. It wasn’t before, per invariant on the `Discoveries` struct
(now deleted). Now that the behavior is controlled by an underlying
trait, I figured that it made more sense to be permissive; I’d expect to
get bugs here otherwise. Can clarify contract.